core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1037,6 +1071,32 @@ private void sendNotificationBroadcast(int type, int code) {
         sendBroadcast(intent);
     }
 
+    private void skipIfMetEnding() {
+        Playable playable = mediaPlayer.getPlayable();
+        if (! (playable instanceof FeedMedia)) { return; }
+
+        int duration = getDuration();
+        int remainingTime = duration - getCurrentPosition();
+
+        FeedMedia feedMedia = (FeedMedia) playable;
+        FeedPreferences preferences = feedMedia.getItem().getFeed().getPreferences();
+        int skipEnd = preferences.getFeedSkipEnding();
+        if (skipEnd > 0 && remainingTime < skipEnd * 1000) {
+            Log.d(TAG, "skipIfMetEnding: Skipping the remaining " + remainingTime / 1000 + " " + skipEnd );
+            Context context = getApplicationContext();
+            String skipMesg = String.format(context.getString(R.string.pref_feed_skip_ending_toast),
+                    skipEnd,
+                    context.getString(R.string.time_seconds));
+            Toast toast = Toast.makeText(context, skipMesg,
+                    Toast.LENGTH_LONG);
+            toast.show();
+
+            mediaPlayer.skip();
```

Comment:
Just wondering: Does this honor the "Follow queue" setting?

Reply:
I tested it and this behaves the same as hitting the 'Skip' button
1) The location is saved
2) The video is skipped and the next one is started.

I don't see 'Follow queue'  option

Reply:
> The video is skipped and the next one is started.

What if you uncheck the option `Follow queue` in Settings » Playback? It should then not start the next file.

Reply:
fixed now and tested.